### PR TITLE
Fixes for StrictEmailAddressAttribute

### PR DESCRIPTION
**Pull request by eliykat**
Fixing QA feedback per this ticket: https://app.asana.com/0/1169444489336079/1200594423596249/f

### CODE REVIEW

**Review by addisonbeck — APPROVED** [review]

File: src/Core/Utilities/StrictEmailAddressAttribute.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public override bool IsValid(object value)
                 return false;
             }
 
-            var illegalChars = @"[\s<>()]";
+            var illegalChars = @"[\s<>();]|[.,]$";
```
Comment (MGibson1) [inline src/Core/Utilities/StrictEmailAddressAttribute.cs:21]: It seems like this is going to end up being whack-a-mole. 

Does is make sense just using something like this:
```c#
System.Net.Mail.MailAddress(emailAddress).Address == emailAddress
```
?

Comment (MGibson1) [reply src/Core/Utilities/StrictEmailAddressAttribute.cs:21]: This still doesn't catch the trailing `.`, btw

Comment (eliykat) [reply src/Core/Utilities/StrictEmailAddressAttribute.cs:21]: That would work great for MailKit, which exposes its own `MailboxAddress.Parse` method - so we could call that and compare the result as you suggest.

However, Amazon SES doesn't seem to expose its own parsing method (at least that I could find), so we couldn't use it there. We'd have to use MailKit or `System.Net.Mail` as analogues and hope that they have similar parsing logic - which they probably do, but then it does feel like we're crossing our fingers. In this case, I think you'd still want the regex there for some certainty (you'd at least need it to capture the trailing `.` if nothing else) - and then that feels like we're doubling up on checks.

What do you think?

Comment (cscharf) [reply src/Core/Utilities/StrictEmailAddressAttribute.cs:21]: I would presume it would be better to apply a general validator from MailKit, and then ALSO validate any illegal characters also rejected upon by SES, using the most restrictive measure.

See: https://stackoverflow.com/a/1903368/14067465

See Also (for SES errors regarding email address): https://docs.aws.amazon.com/ses/latest/DeveloperGuide/troubleshoot-error-messages.html

>**There are non-ASCII characters in the email address** — The email address string must be 7-bit ASCII. If you want to send to or from email addresses that contain Unicode characters in the domain part of an address, you must encode the domain using Punycode. Punycode is not permitted in the local part of the email address (the part before the @ sign) nor in the "friendly from" name. If you want to use Unicode characters in the "friendly from" name, you must encode the "friendly from" name using MIME encoded-word syntax, as described in [Sending raw email using the Amazon SES API](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-raw.html). For more information about Punycode, see [RFC 3492](http://tools.ietf.org/html/rfc3492).

**Comment by eliykat** [comment]
Thanks for your feedback everyone. Here's what I've done:
* use MimeKit to parse the email address and then compare it against the input. Return false if different or if any parsing errors.
* turns out MimeKit strips the trailing `.`, so that edge case is handled.
* wrote unit tests for many of the cases we've discussed, including some from the SO thread that @cscharf linked to
* turns out that MimeKit covers all of these, so I've removed the regex. If our unit tests start failing then we can go back to regex crafting.

@cscharf Unicode email addresses are currently permitted by `StrictEmailAddressAttribute`. I know that we don't properly support those at the moment (we need to do punyencoding for AmazonSES) but I also didn't think it was worth banning them here as they don't do any damage either.

**Review by cscharf — APPROVED** [review]
Great tests, thanks!
